front/scripts/i18n-checker.ts

```diff
@@ -0,0 +1,145 @@
+import { mkdtemp, readFile } from 'node:fs/promises';
+import os from 'node:os';
+
+import { glob } from 'glob';
+import * as I18nextParser from 'i18next-parser';
+import { jsonKeyPathList } from 'json-key-path-list';
+import vfs from 'vinyl-fs';
+
+const IGNORE_MISSING: RegExp[] = [
+  // key used by a t function in modules/trainschedule/components/ManageTrainSchedule/helpers/formatConf.ts
+  /translation:errorMessages\..*/,
+  /translation:error/,
+  /translation:default/,
+  /translation:error/,
+  /translation:unspecified/,
+];
+const IGNORE_UNUSED: RegExp[] = [
+  /.*-generated$/,
+  /.*\.generated\..*$/,
+  /errors:.*/, // Errors are generated and used dynamicly
+  /infraEditor:.*/, // Translation of properties object for the form
+  /infraEditor:__main____.*/, // Found by error by i18n parser in a json-schema
+  /translation:Editor\.tools\..*/, // Editor tool's label are generated
+  /translation:Editor.obj-types\..*/, // Type of object are translated dynamicly on the sumpup popin
+  /translation:Editor.directions\..*/,
+  /translation:Editor.layers\..*/,
+  /Editor\.item-statuses\..*/,
+  /translation:Editor\.infra-errors\.error-type\..*/, // Infra error types are generated
+  /translation:Editor\.infra-errors\.error-level\..*/, // Infra error level are generated
+  /translation:Editor\.infra-errors\.corrector-modal\..*/,
+  /home\/navbar:language\..*/, // Language selector which is generated with the locale
+];
+
+/**
+ * Read a file and returns its content as a JSON
+ */
+async function readJsonFile<T extends { [key: string]: unknown }>(filePath: string): Promise<T> {
+  const data = await readFile(filePath, 'utf-8');
+  return JSON.parse(data);
+}
+
+/**
+ * Given a locales folder, return the list of all i18n keys.
+ */
+async function getLocalesKeys(localePath: string): Promise<Set<string>> {
+  const pathForFrenchLng = `${localePath}/fr/`;
```

Comment:
Do you think we can do the same for english keys ?

Reply:
Yes we can, but from what I understood about the i18n lifecycle is that the developer **has to** manage the i18n keys in `FR`, and **may** do it for `EN`.

Reply:
Solved